timeseries/src/timeseries.rs

```diff
@@ -0,0 +1,143 @@
+//! Core TimeSeries implementation with write API.
```

Comment:
should this just replace `tsdb.rs` or are you planning on using that internally here?